### DISCUSSION TITLE
selfhosted vaultwarden and chrome red page !!!!!

### DISCUSSION THREAD

**Discussion by mauroruffino** [dani-garcia/vaultwarden#5116] [Q&A]
hi everybody,
i'm new in this community and new in vaultwarden environment / nginx environment & friends ;-(

I have a selfhosted vaultwarden docker installation (in  a zimaboard casaos) that i cannot open from browser  because https is disabled. 
if i use an external domain (e.g. myexternaldomain.duckdns.org) with nginx reverse proxy ((with letsencrypt cert)) that forward https:\\myexternaldomain.duckdns.org url to my local url http:\\casaos.local:8888 

all works fine if i use edge browser or firefox browser but if i use chrome browser, a great red page the says me that server is dangerous!!!!!

no problem from chrome extension from bitwarden android app or  from windows bitwarden standalone client app, but with chrome browser ---> red page

can anyone helps me ?
thanks
bye
mauro

**Comment by etymotic** [comment]
What happens if you use the local IP address:8888 instead of casaos.local:8888?

Is your DNS server (router or whatever you have set up) rewriting the DNS entry for myexternaldomain.duckdns.org so it doesn't "leave" your LAN? 

If not, do you have anything configured in your router for Hairpin NAT? The fact that it's working with other browsers makes me think that's not the problem, but just figured I'd ask.

**Reply by mauroruffino** [reply]
i dont know i'll verify in the next days. if i use ipaddress:8888 its the same instead casaos.local:8888

the chrome red page its showed only from chrome browser and noone warning from other browser

but if i use ipaddress:8888 or casaos.local:9999 from every address....

the login page its ok (http://casaos.local:8888/#/login) also the page where i type main password.... the warning about https not enabled is showed AFTER then i click on the blue bottom to enter ....
![Screenshot_20241021_081607](https://github.com/user-attachments/assets/5562458f-fc91-4d00-b9a6-118dc3d241a8)
![Screenshot_20241021_081540](https://github.com/user-attachments/assets/68317a21-1926-4cfe-a654-cea127947661)

**Reply by BlackDex** [reply]
The problem is that browsers do not allow the Crypto API to be accessed for sites that do not use HTTPS, unless they are running on localhost/127.0.0.1. So that is why you receive that message and it doesn't work.

If you use anything else besides localhost/127.0.0.1 you need a valid certificate or one that is accepted by the browser once you approved it.

**Reply by mauroruffino** [reply]
from external with letsencrypt certificate works but with chrome : red page, with firefox , edge no red page and works perfectly...

i read about https but seems a little bit difficult to enable it

**Reply by etymotic** [reply]
Perhaps it's something with Nginx, but if it works from outside the network, then maybe not.

If nginx is too difficult to figure out (it is for me), try Nginx Proxy Manager instead. It simplifies everything and has a web GUI.

**Reply by mauroruffino** [reply]
Yes @etymotic  i use npm and it works , until yesterday works but with the chrome red page (something like that https://www.reddit.com/r/selfhosted/comments/ut1rhk/wtf_google_marked_my_personal_site_phishing/ )

and if i go to my site from "visti this unsafe site" , my vaultwarden works

TODAY chrome red page is disappeared!!!!, i dont know why!!!! and all works good , from external url
from internal url nothing is changed .... but for me if chrome is not angry with red page, im happy and for me is good

what do you think ? i reach my selfhosted from .....duckdns.org with a lets encrypt certificate , then i think that is enough
![Screenshot_20241021_220759](https://github.com/user-attachments/assets/da00b931-ddc9-48dd-b23f-eabf5f11e955)
